### PR TITLE
Layer Check updates

### PR DESCRIPTION
**Pull request by markfields**
* Raise an error if a cyclical dependency is introduced between layers (meaning package in A depends on package in B, and some other package in B depends on some other package in A).
* ~Run layer-check to generate PACKAGES.md during postinstall step~ See https://github.com/microsoft/FluidFramework/pull/8173
* Some refactoring

### CODE REVIEW

**Comment by markfields** [comment]
@sumedhb1995 - with this change the error you would face in your PR #8085 would be:

```
ERROR: Circular dependency detected between layers!
Please inspect this subset of layers and their dependencies to find the cycle:

HostUtils --> Test
Examples --> HostUtils,Test
Tools --> Test
Test --> Tools

Note that the dependency may not be explicit in the layer info JSON,
and doesn't indicate a strict circular dependency between packages.
But some packages in layer A depend on packages in layer B, and likewise some in B depend on some in A.
```

**Comment by andre4i** [comment]
That's awesome, @markfields ! This will also fix https://github.com/microsoft/FluidFramework/issues/6512

**Comment by markfields** [comment]
@andre4i I don't think this will fix that issue...  This fix is just about the layer level, not individual packages.

**Comment by andre4i** [comment]
That's correct, you're right. But the first part of my comment still stands!

File: tools/build-tools/src/layerCheck/layerGraph.ts [hunk 1]
```diff
@@ -497,9 +497,9 @@ export class LayerGraph {
             }
 
             this.padArraysToSameLength(packagesInCell, layersInCell, "&nbsp;");
-            lines.push(`| Packages | Layer Dependencies |`);
-            lines.push(`| --- | --- |`);
-            lines.push(`| ${packagesInCell.join("</br>")} | ${layersInCell.join("</br>")} |${newline}`);
+           lines.push(`| Packages | Layer Dependencies |`);
+           lines.push(`| --- | --- |`);
+           lines.push(`| ${packagesInCell.join("</br>")} | ${layersInCell.join("</br>")} |${newline}`);
```
Comment (markfields) [inline tools/build-tools/src/layerCheck/layerGraph.ts:502]:
```suggestion
            lines.push(`| Packages | Layer Dependencies |`);
            lines.push(`| --- | --- |`);
            lines.push(`| ${packagesInCell.join("</br>")} | ${layersInCell.join("</br>")} |${newline}`);
```

**Comment by markfields** [comment]
I don't know why `PACKAGES.md` is ending up modified in the CI (which fails the build)... I've merged main and not getting any changes to the file when I run layer-check...

**Review by andre4i — APPROVED** [review]

**Pull request merged**